Make blows up with missing curl/curl.h

**Describe the bug**
While compiling server with `./configure --disable-client --disable-manager --enable-fcgi && make` on a system that does not have libcurl installed, configure does not catch the missing dependency and the build blows up.

```
  CXX      remote_submit.o
../lib/remote_submit.cpp:24:10: fatal error: curl/curl.h: No such file or directory
 #include <curl/curl.h>
          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:820: remote_submit.o] Error 1
```

**Steps To Reproduce**
```
git clone https://github.com/BOINC/boinc.git boinc-git
cd boinc-git
git checkout tags/server_release/1.2/1.2.1 -b server_release/1.2.1
./_autosetup
./configure --disable-client --disable-manager --enable-fcgi
make
```

**Expected behavior**
Configure should notify about the missing lib and abort.

--- a/configure.ac
+++ b/configure.ac
@@ -451,7 +451,7 @@ AS_IF([test "x${enable_client}" = "xyes" -o "x${enable_server}" = "xyes"],
 dnl ---------- libcurl (m4/libcurl.m4) ------------------------------
 dnl curl is needed for client
 
-AS_IF([test "x${enable_client}" = "xyes"],
+AS_IF([test "x${enable_client}" = "xyes" -o "x${enable_server}" = "xyes"],
   [
     LIBCURL_CHECK_CONFIG([yes], [7.17.1], [haveCurl=yes], [haveCurl=no])
 

